tests/modules/lib/cddl-gen/CMakeLists.txt

```diff
@@ -12,7 +12,21 @@ project(test_cddl_gen)
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
 
-target_cddl_source(app test.cddl ENTRY_TYPES Test ENCODE
-  ${VERBOSE}
-  ${CANONICAL}
+set(py_command
+  cddl-gen
+  -c ${CMAKE_CURRENT_LIST_DIR}/test.cddl
+  code
+  --output-cmake ${PROJECT_BINARY_DIR}/test.cmake
+  -t Test
+  -e
   )
+
+execute_process(
+  COMMAND ${py_command}
+  COMMAND_ERROR_IS_FATAL ANY
+  )
+
+include(${PROJECT_BINARY_DIR}/test.cmake)
+
+target_link_libraries(test PRIVATE zephyr_interface)
```

Comment:
just here you have some inside knowledge of a `test` target.

From where did this knowledge originate ?
And based on what rules are the library name `test` chosen ?
based on the `test.cddl` filename ?

This has a high risk of colliding with another library name chosen by a developer, which again may cause strange CMake errors that a user suddenly must try to understand why happens.